reactions/debezium/debezium-reaction/Services/ChangeHandler.cs

```
@@ -0,0 +1,112 @@
+		_includeSchemas = config.GetValue<bool?>("includeSchemas") ?? true;
+		_includeKey = config.GetValue<bool?>("includeKey") ?? true;	
+		
+		// A list of brokers; represented as a comma-separated string
+		string brokers = config.GetValue<string>("brokers") ?? throw new ArgumentNullException("Debezium brokers is required");
+
+		ProducerConfig producerConfig = new ProducerConfig
+		{
+			BootstrapServers = brokers
+		};
+
+		_producer = new ProducerBuilder<Null, string>(producerConfig).Build();
+
+		_formatter = new DataChangeEventFormatter(_includeKey, _includeSchemas);
+	}
+
+	public async Task HandleChange(ChangeEvent evt, object? queryConfig)
+	{
+		var metadata = new EventMetadata(evt);
+
+		_logger.LogInformation("Processing {QueryId}", metadata.QueryId);
+
+		await ProcessResults(metadata, evt.AddedResults, "c");
+
+		var updatedResults = new List<Dictionary<string, object>>();
+		for (int i = 0; i < evt.UpdatedResults.Length; i++)
+		{
+			var currResultDict = new Dictionary<string, object>
+			{
+				["before"] = evt.UpdatedResults[i].Before,
+				["after"] = evt.UpdatedResults[i].After
+			};
+			updatedResults.Add(currResultDict);
+		}
+		await ProcessResults(metadata, [.. updatedResults], "u");
+		await ProcessResults(metadata, evt.DeletedResults, "d");
+
+		_producer.Flush();
+	}
+
+	private async Task ProcessResults(EventMetadata metadata, Dictionary<string, object>[] results, string op)
+	{
+		var noIndent = new JsonSerializerOptions { WriteIndented = false };
+		foreach (var res in results)
+		{
+			string eventString = _formatter.FormatEvent(metadata, res, op);
+
+			_logger.LogInformation($"dataChangeEvent: {eventString}");
+			try
+			{
+				var deliveryReport = await _producer.ProduceAsync(_topic, new Message<Null, string> { Value = eventString }, CancellationToken.None);
+
+				if (deliveryReport.Status != PersistenceStatus.Persisted)
+				{
+					_logger.LogInformation($"Delivery failed: {deliveryReport.Message.Value}");
```

If the delivery failed, should we throw an exception, so that the message from the dapr queue in not acknowledged, and it can retry?